File: TP2/TP2/src/main/java/SistemaImobiliaria/Imovel.java

package SistemaImobiliaria;

public class Imovel implements Comparable <Imovel>{
    private int id, numero, quartos, banheiros;
    private double valor;
    private String proprietario, rua, bairro, cidade;
    
    //Sobrecarga do <<
    @Override
    public String toString() {
        return "-----\nProprietario: " + proprietario + "\nValor do Imovel: "  + valor + "\nNumero de Quartos: " + quartos + "\nRua: " + rua + "\nBairro: " + bairro + "\nCidade: " + cidade + "\n";
    }
    
    //Construtor de Imovel vazio
    public Imovel() {
        id = numero = quartos = banheiros = 0;
        valor = 0;
        proprietario = "";
        rua = "";
        bairro = "";
        cidade = "";
    }
    
    //Construtor de imovel
    public Imovel(int id, int numero, int quartos, int banheiros, double valor, String proprietario, String rua, String bairro, String cidade) {
        this.id = id;
        this.numero = numero;
        this.quartos = quartos;
        this.banheiros = banheiros;
        this.valor = valor;
        this.proprietario = proprietario;
        this.rua = rua;
        this.bairro = bairro;
        this.cidade = cidade;
    }
    //Getters e Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public int getQuartos() {
        return quartos;
    }

    public void setQuartos(int quartos) {
        this.quartos = quartos;
    }

    public int getBanheiros() {
        return banheiros;
    }

    public void setBanheiros(int banheiros) {
        this.banheiros = banheiros;
    }

    public double getValor() {
        return valor;
    }

    public void setValor(double valor) {
        this.valor = valor;
    }

    public String getProprietario() {
        return proprietario;
    }

    public void setProprietario(String proprietario) {
        this.proprietario = proprietario;
    }

    public String getRua() {
        return rua;
    }

    public void setRua(String rua) {
        this.rua = rua;
    }

    public String getBairro() {
        return bairro;
    }

    public void setBairro(String bairro) {
        this.bairro = bairro;
    }

    public String getCidade() {
        return cidade;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }
    
    //Implementacao da funcao compareTo, que implementa a
    //interface Comparable
    @Override
    public int compareTo(Imovel o) {
        if(valor > o.valor)
            return 1;
        else if(valor == o.valor)
            return 0;
        else
            return -1;
    }
    
}

File: TP01/imovel.cpp
#include <iostream>
#include <string>
#include "imovel.h"

using namespace std;
//Construtor da classe imovel
Imovel::Imovel(int id, int numero, int quartos, int banheiros, float valor, string proprietario, string rua, string bairro, string cidade): 
    id(id), numero(numero) , quartos(quartos) , banheiros(banheiros), valor(valor), proprietario(proprietario), rua(rua), bairro(bairro), cidade(cidade){}

Imovel:: ~Imovel(){}
//GETTERS E SETTERS
int Imovel::getId(){
    return id;
}

void Imovel::setId(int n){
    id = n;
}

int Imovel::getNumero(){
    return numero;
}

void Imovel::setNumero(int n){
    numero = n;
}

int Imovel::getQuartos(){
    return quartos;
}

void Imovel::setQuartos(int n){
    quartos = n;
}

int Imovel::getBanheiros(){
    return banheiros;
}

void Imovel::setBanheiros(int n){
    banheiros = n;
}

float Imovel::getValor(){
    return valor;
}

void Imovel::setValor(float v){
    valor = v;
}

string Imovel::getProprietario(){
    return proprietario;
}

void Imovel::setProprietario(string n){
    proprietario = n;
}

string Imovel::getRua(){
    return rua;
}

void Imovel::setRua(string n){
    rua = n;
}

string Imovel::getBairro(){
    return bairro;
}

void Imovel::setBairro(string n){
    bairro = n;
}

string Imovel::getCidade(){
    return cidade;
}

void Imovel::setCidade(string n){
    cidade = n;
}
//SOBRECARGA DO OPERADOR <<
void Imovel::saida(ostream& out) const{
    out << "-----\nProprietario: " << proprietario << "\nValor do Imovel: " << valor << "\nNumero de Quartos: " << quartos << "\nRua: "
    << rua << "\nBairro: " << bairro << "\nCidade: " << cidade << "\n";
}

ostream& operator<<(ostream& out, const Imovel& obj){
    obj.saida(out);
    return out;
}
